src/frontend-build.sh

```diff
@@ -1,240 +1,216 @@
-export LC_ALL=en_US.utf-8
-export LANG=en_US.utf-8
-export APP_ROOT=${APP_ROOT:-pwd}
-export WORKSPACE=${WORKSPACE:-$APP_ROOT}  # if running in jenkins, use the build's workspace
-# export IMAGE_TAG=$(git rev-parse --short=7 HEAD)
-export GIT_COMMIT=$(git rev-parse HEAD)
+    # Get the latest git commit hash
+    export GIT_COMMIT=$(git rev-parse HEAD)
+}
 
+load_cicd_helper_functions() {
+    local LIBRARY_TO_LOAD="$1"
+    local CICD_TOOLS_REPO_BRANCH='main'
+    local CICD_TOOLS_REPO_ORG='RedHatInsights'
+    local CICD_TOOLS_URL="https://raw.githubusercontent.com/${CICD_TOOLS_REPO_ORG}/cicd-tools/${CICD_TOOLS_REPO_BRANCH}/src/bootstrap.sh"
+    source <(curl -sSL "$CICD_TOOLS_URL") "$LIBRARY_TO_LOAD"
+}
 
-if [[ -z "$QUAY_USER" || -z "$QUAY_TOKEN" ]]; then
-    echo "QUAY_USER and QUAY_TOKEN must be set"
-    exit 1
-fi
+# Detect if this is a PR build and set the appropriate variables
+set_pr_details() {
+  # If the job name doesn't contain "pr-check", exit early
+  if ! echo "$JOB_NAME" | grep -w "pr-check" > /dev/null; then
+    return
+  fi
 
-if [[ -z "$RH_REGISTRY_USER" || -z "$RH_REGISTRY_TOKEN" ]]; then
-    echo "RH_REGISTRY_USER and RH_REGISTRY_TOKEN must be set"
-    exit 1
-fi
-
-# Chrome isn't currently using our config, don't need this complexity for now
-# Not deleting, as we may need to re-enable later
-#if [ $APP_NAME == "chrome" ] ; then
-  # get_chrome_config;
-#fi
-
-DOCKER_CONFIG="$PWD/.docker"
-mkdir -p "$DOCKER_CONFIG"
-echo $QUAY_TOKEN | docker  login -u="$QUAY_USER" --password-stdin quay.io
-echo $RH_REGISTRY_TOKEN | docker  login -u="$RH_REGISTRY_USER" --password-stdin registry.redhat.io
-
-pwd
-ls -alsvh
-cat Dockerfile
-
-#PRs shouldn't get the special treatment for history
-if [ $IS_PR = true ]; then
-  docker  build -t "${IMAGE}:${IMAGE_TAG}" $APP_ROOT -f $APP_ROOT/Dockerfile
-  docker  push "${IMAGE}:${IMAGE_TAG}"
-  teardown_docker
-else
-  # Build and push the -single tagged image
-  # This image contains only the current build
-  docker  build --label "image-type=single" -t "${IMAGE}:${IMAGE_TAG}-single" $APP_ROOT -f $APP_ROOT/Dockerfile
-  docker  push "${IMAGE}:${IMAGE_TAG}-single"
+  local timestamp=$(date +%s)
+  local pr_id="unknown_pr_id"
+  
+  if [ -n "$ghprbPullId" ]; then
+    pr_id=$ghprbPullId
+  elif [ -n "$gitlabMergeRequestIid" ]; then
+    pr_id=$gitlabMergeRequestIid
+  fi
 
-  # Get the the last 6 builds
-  getHistory
+  export IMAGE_TAG="pr-${pr_id}-${IMAGE_TAG}"
+  export CONTAINER_NAME="${APP_NAME}-pr-check-${pr_id}-${timestamp}"
+  export IS_PR=true
+}
 
-  # Build and push the aggregated image
-  # This image is tagged with just the SHA for the current build
-  # as this is the one we want deployed
-  docker  build --label "image-type=aggregate" -t "${IMAGE}:${IMAGE_TAG}" $APP_ROOT -f $APP_ROOT/Dockerfile
-  docker  push "${IMAGE}:${IMAGE_TAG}"
+main() {
+  # Load the CICD helper scripts
+  load_cicd_helper_functions container
+  load_cicd_helper_functions image_builder
```

Comment:
image_builder already loads `container` so you can leave `image_builder` alone

Reply:
Removed in recent push